test/jdk/javax/swing/UIDefaults/6302464/bug6302464.java

```diff
@@ -118,28 +118,46 @@ private static void testFontRenderingContext(Object aaHint) {
     private static void testAntialiasingHints() {
         setMetalLookAndFeel();
 
-        HashSet colorsAAOff = getAntialiasedColors(VALUE_TEXT_ANTIALIAS_OFF, 100);
+        boolean isMacOSX14 = false;
+        if (System.getProperty("os.name").contains("OS X")) {
+            String version = System.getProperty("os.version", "");
+            if (version.startsWith("10.")) {
+                version = version.substring(3);
+                int periodIndex = version.indexOf('.');
+                if (periodIndex != -1) {
+                    version = version.substring(0, periodIndex);
+                }
+                try {
+                    int v = Integer.parseInt(version);
+                    isMacOSX14 = (v >= 14);
+                } catch (NumberFormatException e) {
+                }
+            }
+        }
+        if (!isMacOSX14) {
+            HashSet colorsAAOff = getAntialiasedColors(VALUE_TEXT_ANTIALIAS_OFF, 100);
```

Comment:
what happens on macOS 11 ?

Reply:
it also works in macos11

Reply:
So on macOS 11 we enter this block since isMacOSX14 is false and the test passes ?

Reply:
No, it still fails in macosx11 without this modification. macos11, System .getProperty returns 10.16 so it satisfies this macosx >=10.14 condition and it bails out from this subtest.

Reply:
Are you sure you are up to date ? There was a fix so that it returns 11

My build on my machine :
~/jdk.git/build/macosx-x86_64-server-release/jdk/bin/jshell
|  Welcome to JShell -- Version 17-internal
|  For an introduction type: /help intro

jshell> System.getProperty("os.version")
$1 ==> "11.2.3"

Reply:
Yes, the repo was a bit outdated on BigSur system. I have updated the fix to check BigSur OS too.